File: cpp_on_sea/adler32/typescript/adler32.ts
// adler32_checksum.ts
// Computes Adler-32 checksum for a randomly generated list of 50 million unsigned 8-bit integers

function generateRandomBytes(length: number): Uint8Array {
  const buffer = new Uint8Array(length);
  for (let i = 0; i < length; i++) {
    buffer[i] = Math.floor(Math.random() * 256);
  }
  return buffer;
}

function adler32(data: Uint8Array): number {
  const MOD_ADLER = 65521;
  let a = 1;
  let b = 0;
  for (let i = 0; i < data.length; i++) {
    a = (a + data[i]) % MOD_ADLER;
    b = (b + a) % MOD_ADLER;
  }
  return (b << 16) | a;
}

const BYTE_COUNT = 50_000_000;
console.log(`Generating ${BYTE_COUNT.toLocaleString()} random bytes...`);
const bytes = generateRandomBytes(BYTE_COUNT);
console.log(`Computing Adler-32 checksum...`);
const checksum = adler32(bytes);
console.log(`Adler-32 checksum = 0x${checksum.toString(16).padStart(8, "0")}`);

File: cpp_on_sea/adler32/ruby/adler32.rb
# adler32_checksum.rb

# This Ruby script generates a specified number of random unsigned 8-bit integers
# and computes the Adler-32 checksum over the resulting data buffer using a manual implementation.

# Adler-32 parameters
MOD_ADLER = 65_521

# Function to compute Adler-32 checksum manually
# num_bytes - number of random bytes (0..255) to generate
# Returns the Adler-32 checksum as an Integer

def compute_adler32(num_bytes)
  a = 1
  b = 0
  rand_gen = Random.new

  # Generate the full data buffer
  data = rand_gen.bytes(num_bytes)

  # Process entire buffer in one loop
  data.each_byte do |byte|
    a = (a + byte) % MOD_ADLER
    b = (b + a) % MOD_ADLER
  end

  # Combine sums into final checksum
  (b << 16) | a
end

# Main execution block
if __FILE__ == $0
  # Default number of bytes: 50 million
  num_bytes = 50_000_000

  puts "Generating \#{num_bytes} random 8-bit unsigned integers and computing Adler-32 checksum..."

  checksum = compute_adler32(num_bytes)

  puts "\nAdler-32 checksum: \#{checksum}"

end
s